src/quiz/use-quiz.ts

```diff
@@ -1,23 +1,70 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
 
 import { type Question } from "./types";
 
-export const useQuiz = (initialQuestions: Question[]) => {
+interface Props {
+	initialQuestions: Question[];
+	validationMessages: {
+		correct: string;
+		incorrect: string;
+	};
+}
+
+export const useQuiz = ({ initialQuestions, validationMessages }: Props) => {
 	const [quizAnswers, setQuizAnswers] = useState<(number | undefined)[]>(
 		initialQuestions.map((question) => question.selectedAnswer),
 	);
+	const [questions, setQuestions] = useState<Question[]>([]);
+	const [correctAnswerCount, setCorrectAnswerCount] = useState(0);
```

Comment:
Either is fine, but I lean towards `number`, because that makes more sense with the following logic: if validated, then render count, else not.

Also, `if (count !== null) ...` is a little ugly, imo.